From 8e75ddeac6f282eb7bb5e3028db633944807a1f0 Mon Sep 17 00:00:00 2001
From: Colin Walters <walters@verbum.org>
Date: Wed, 08 Sep 2010 14:57:11 +0000
Subject: introspection: Update to new scanner API

* Add --warn-all
* Drop unnecessary  of --strip-prefix; for GdkX11, use --symbol-prefix

https://bugzilla.gnome.org/show_bug.cgi?id=628160

Conflicts:

	configure.in
	gdk/Makefile.am
	gtk/Makefile.am
---
diff --git a/configure.in b/configure.in
index fb5a24f..4bf26d7 100644
--- a/configure.in
+++ b/configure.in
@@ -1565,7 +1565,7 @@ fi
 # GObject introspection
 ##################################################
 
-GOBJECT_INTROSPECTION_CHECK([0.6.7])
+GOBJECT_INTROSPECTION_CHECK([0.9.5])
 
 ##################################################
 # Checks for gtk-doc and docbook-tools
diff --git a/gdk/Makefile.am b/gdk/Makefile.am
index c7d135e..11843a2 100644
--- a/gdk/Makefile.am
+++ b/gdk/Makefile.am
@@ -196,7 +196,7 @@ introspection_files = 		\
         gdkenumtypes.h
 
 Gdk-2.0.gir: $(gdktargetlib) Makefile
-Gdk_2_0_gir_SCANNERFLAGS = --strip-prefix=Gdk
+Gdk_2_0_gir_SCANNERFLAGS = --warn-all
 Gdk_2_0_gir_INCLUDES = Gio-2.0 GdkPixbuf-2.0 Pango-1.0 cairo-1.0
 Gdk_2_0_gir_LIBS = $(gdktargetlib)
 Gdk_2_0_gir_FILES = $(introspection_files)
@@ -237,7 +237,7 @@ x11_introspection_files = 		\
         x11/gdkx.h
 
 GdkX11-2.0.gir: $(gdktargetlib) Gdk-2.0.gir Makefile
-GdkX11_2_0_gir_SCANNERFLAGS = --strip-prefix=gdk_x11
+GdkX11_2_0_gir_SCANNERFLAGS = --warn-all --identifier-prefix=Gdk --symbol-prefix=gdk
 GdkX11_2_0_gir_INCLUDES = Gio-2.0 Gdk-2.0 GdkPixbuf-2.0 Pango-1.0 xlib-2.0
 GdkX11_2_0_gir_LIBS = $(gdktargetlib)
 GdkX11_2_0_gir_FILES = $(x11_introspection_files)
diff --git a/gtk/Makefile.am b/gtk/Makefile.am
index a191d6d..18667f1 100644
--- a/gtk/Makefile.am
+++ b/gtk/Makefile.am
@@ -991,7 +991,10 @@ introspection_files = \
     gtktypebuiltins.c
 
 Gtk-2.0.gir: $(INTROSPECTION_SCANNER) $(gtktargetlib) $(top_builddir)/gdk/Gdk-2.0.gir Makefile
-Gtk_2_0_gir_SCANNERFLAGS = --strip-prefix=Gtk
+Gtk_2_0_gir_SCANNERFLAGS = --warn-all --add-include-path=$(top_builddir)/gdk
+if USE_X11
+Gtk_2_0_gir_SCANNERFLAGS += --add-include-path=$(top_builddir)/gdk/x11
+endif
 Gtk_2_0_gir_INCLUDES = Atk-1.0 Gdk-2.0
 Gtk_2_0_gir_CFLAGS = \
 		$(INCLUDES) \
--
cgit v0.8.3.1
